ddtrace/_trace/sampling_rule.py

```diff
@@ -153,26 +153,25 @@ def check_tags(self, meta, metrics):
         tag_match = False
         for tag_key in self._tag_value_matchers.keys():
             value = meta.get(tag_key)
-            tag_match = self._tag_value_matchers[tag_key].match(str(value))
-            # If the value doesn't match in meta, check the metrics
-            if tag_match is False:
+            # it's because we're not checking metrics first before continuing
+            if value is None:
                 value = metrics.get(tag_key)
+                if value is None:
+                    continue
                 # Floats: Matching floating point values with a non-zero decimal part is not supported.
                 # For floating point values with a non-zero decimal part, any all * pattern always returns true.
                 # Other patterns always return false.
                 if isinstance(value, float):
                     if not value.is_integer():
-                        if self._tag_value_matchers[tag_key].pattern == "*":
+                        if all(c == "*" for c in self._tag_value_matchers[tag_key].pattern):
```

Comment:
Here we expanding support for regex patterns from `*` to `*`x N times. Do we need this change?